core/include/masks/annulus.hpp

```diff
@@ -0,0 +1,150 @@
+/** Detray library, part of the ACTS project (R&D line)
+ * 
+ * (c) 2020 CERN for the benefit of the ACTS project
+ * 
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+#include "masks/mask_identifier.hpp"
+#include "core/intersection.hpp"
+#include "utils/containers.hpp"
+#include "tools/planar_intersector.hpp"
+
+#include <cmath>
+#include <climits>
+
+namespace detray
+{
+    /** This is a 2-dimensional mask for the annulus geometry that is
+     *  e.g. used for the itk strip endcaps.
+     * 
+     * It is defined by the two radii _values[0] and  _values[1] in the polar
+     * coordinate system of and endcap strip module, as well as the two phi 
+     * boundaries, _values[2] and _values[3], that are only easily definable in 
+     * the local strips system (relative to the average Phi along the 
+     * strips r coordinate). Using a conversion between the two coordinate 
+     * systems, these boundaries can be checked with a tolerance in r (t0 and 
+     * t1), as well as phi (t3 and t3). 
+     * Due to the local polar coordinate system of the strips needing a 
+     * different origin from the discs polar one, three additional conversion
+     * parameters are included (_values[4], values[5], _values[6]).
+     * Here, the first two are the origin shift in xy, while _values[6] is the 
+     * average Phi angle mentioned above.
+     **/
+    template <typename scalar_type,
+              typename intersector_type = planar_intersector,
+              typename links_type = bool,
+              unsigned int kMaskIdentifier = e_annulus>
+    struct annulus
+    {
+        using mask_values = darray<scalar_type, 7>;
+
+        mask_values _values = {0., std::numeric_limits<scalar_type>::infinity(),
+                               -std::numeric_limits<scalar_type>::infinity(), 
+                               std::numeric_limits<scalar_type>::infinity(),
+                               0., 0., 0.};
+
+        links_type _links;
+
+        static constexpr unsigned int mask_identifier = kMaskIdentifier;
+
+        /** Assignment operator from an array, convenience function
+         * 
+         * @param rhs is the right hand side object
+         **/
+        annulus<scalar_type, intersector_type, links_type, kMaskIdentifier> &
+        operator=(const darray<scalar_type, 7> &rhs)
+        {
+            _values = rhs;
+            return (*this);
+        }
+
+        /** Mask operation 
+         * 
+         * @tparam point2_type is the type of the point to be checked w.r.t. to
+         * the mask bounds
+         * 
+         * @param p the point to be checked in local polar coord
+         * @param t0 is the tolerance in minR
+         * @param t1 is the tolerance in maxR
+         * @param t2 is the tolerance in minPhi
+         * @param t2 is the tolerance in maxPhi
+         * 
+         * @return an intersection status e_inside / e_outside
+         **/
+        template <typename point2_type>
```

Comment:
Did you not change the `mask` API from `operator()` to `is_inside()` in the last PR.
Am I missing something?